python/lsst/ctrl/mpexec/cli/opt/options.py

```diff
@@ -22,6 +22,38 @@
 import click
 
 from lsst.daf.butler.cli.utils import MWOptionDecorator, MWPath, split_commas, unwrap
+from lsst.daf.butler.core.utils import iterable
+
+
+def _split_commas_int(context, param, values):
```

Comment:
Why has this been copied from daf_butler? Don't we have the ability to use the daf_butler callback?

cc/ @n8pease

Reply:
It's not an exact copy, daf_butler's one splits things into strings, this one make integers instead. But I should have probably called `split_commas` from this method instead of re-implementing its logic.

Reply:
Re-reading the question again - I guess we could update `split_commas` to do type conversion, but my general problem is that `click` callbacks do not seem to be a good match for that problem. AFAIK callbacks happen after type conversion is already applied to argument, and in our case of comma-separated lists it cannot even work. What I mean is that if you declare an option to have type `int` than you cannot pass comma-separated string to it because it will try to convert that string to `int` and will fail. My current solution is not to declare type of an option (essentially making it `str`) but cheat in the callback and return integers from that. I don't know if this is even allowed, docs don't say if this is acceptable, so there is no guarantee that it will not break.

Reply:
Just for record - I reimplemented my `_split_commas_int` on top of `split_commas`, for better unification of the latter we may need some thinking, so I want to keep `_split_commas_int` here for now.